### DISCUSSION TITLE
Timing of test using dayjs/moment

### DISCUSSION THREAD

**Discussion by geekiam23** [cypress-io/cypress#14208] [Component Testing]
Refers to running in the CI/CD

In our test suite, we have tests that start a service at a certain time with moment/dayjs. Running into a problem with the timing and looking to make it more reliable. If a test fails or if the computer does not perform correctly, it causes the test to fail. One problem that I notice is that moment/dayjs sets the time when you give the command instead of when it gets to each file (I know that's how JS functions). 

Ex: Test 1 has a time of 2 mins from current time. Test 6 has a time of 7 mins from current time.

Let me know if this make sense or not. Would love to give more details.

I have one idea but do not know if it is best practice:
Instead of running the entire test suite, I can stagger the timing on each test file and only have to worry about the timing conflict of each file instead of the entire suite.

**Comment by jennifer-shehane** [comment]
Have you looked into using `cy.clock()` to mock the Dates? https://on.cypress.io/clock

**Reply by geekiam23** [reply]
I checked it out but would love to know more about your thoughts. Do you have any examples that you an share with me?
